Remove AT_LINK_STYLE entirely.

ezyang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #29731 Split libtorch.so back into libtorch_{cpu,cuda,hip}
* #29730 Refactor target_compile_options into torch_compile_options
* **#29729 Remove AT_LINK_STYLE entirely.**

It already errored when you built with CUDA/HIP support as no longer supported;
now I expunge it entirely.  Along the way, I delete useless INTERFACE
libraries (which aren't used anywhere else in the cmake.)

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D18571167](https://our.internmc.facebook.com/intern/diff/D18571167)

ezyang committed:
Remove AT_LINK_STYLE entirely.

It already errored when you built with CUDA/HIP support as no longer supported;
now I expunge it entirely.  Along the way, I delete useless INTERFACE
libraries (which aren't used anywhere else in the cmake.)

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

kostmo commented:
## CircleCI build failures summary
As of commit 25c81bd:
* **1/1** failures introduced in this PR
* **0/1** recognized as flaky



Here are the [reasons each build failed](https://dr.pytorch.org/commit-details.html?sha1=25c81bdac8695799af7ccab7e3eae9e3ad4ef1f8):

| Job | Step | Log excerpt |
| --- | --- | --- |
| pytorch_xla_linux_xenial_py3_6_clang7_build | Build | `Failed to run '['/var/lib/jenkins/workspace/xla/scripts/generate_code.sh']'` |


---
This comment was automatically generated by [Dr. CI](https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment).
Follow [this link to opt-out](https://dr.pytorch.org/admin/comments-opt-out.html) of these comments for your Pull Requests.


Please report bugs/suggestions on the [GitHub issue tracker](https://github.com/kostmo/circleci-failure-tracker/issues).

ezyang committed:
Update on "Remove AT_LINK_STYLE entirely."


It already errored when you built with CUDA/HIP support as no longer supported;
now I expunge it entirely.  Along the way, I delete useless INTERFACE
libraries (which aren't used anywhere else in the cmake.)

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

ezyang committed:
Update on "Remove AT_LINK_STYLE entirely."


It already errored when you built with CUDA/HIP support as no longer supported;
now I expunge it entirely.  Along the way, I delete useless INTERFACE
libraries (which aren't used anywhere else in the cmake.)

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]
